internal/common/baseclient/iito.go

```diff
@@ -2,43 +2,61 @@ package baseclient
 
 import "github.com/SOMAS2020/SOMAS2020/internal/common/shared"
 
-// RequestGift allows clients to signalize that they want a gift
+// GetGiftRequests allows clients to signalize that they want a gift
 // This information is fed to OfferGifts of all other clients.
 // COMPULSORY, you need to implement this method
-func (c *BaseClient) RequestGift() uint {
-	if c.clientGameState.ClientInfo.LifeStatus == shared.Critical {
-		return 100
+func (c *BaseClient) GetGiftRequests() []shared.GiftRequest {
+	requests := []shared.GiftRequest{}
+
+	// FIXME: Is this the right way to get the client statuses?
+	for team, status := range c.clientGameState.ClientLifeStatuses {
+		if status == shared.Critical {
+			requests[team] = shared.GiftRequest{RequestFrom: shared.ClientID(team), RequestAmount: 100.0}
+		} else {
+			requests[team] = shared.GiftRequest{RequestFrom: shared.ClientID(team), RequestAmount: 0.0}
+		}
```

Comment:
While the client now has a serverReadHandle, it isn't clear whether this is deprecated. I couldn't find any code that synchronises the two. Review? @mpardalos @lhl2617

Reply:
Yeah you're right, `baseclient.clientGameState` should be deleted. I have made the changes locally, how should I submit them?

Reply:
Put them in a separate PR, request my review? I'll merge it in and apply it here.